### DISCUSSION TITLE
Is wrapping `Popover` content in a `Dialog` mandatory for accessibility or stability?

### DISCUSSION THREAD

**Discussion by NarendraKuruva** [adobe/react-spectrum#7395] [General]
Hi @snowystinger @devongovett,

I was going through the [React Aria Popover documentation](https://react-spectrum.adobe.com/react-aria/Popover.html#features) and noticed there’s no explicit mention of whether popover content should be wrapped in a Dialog.

In my current project ([CodeSandbox link](https://codesandbox.io/p/sandbox/serverless-worker-hqt8tr?file=%2Fsrc%2FTesting.tsx)), I’m rendering content inside the Popover but not wrapping it in a Dialog. Everything seems to work fine.

However, I’d like to confirm:
1. Is it mandatory to use a Dialog for Popover content?
2. Are there any accessibility issues or potential bugs if I skip wrapping the content in a Dialog?

Thank you for your guidance!

**Comment by RajaTheKing826** [comment]
yah yah i too got this doubt while im using react-aria-components

**Comment by snowystinger** [comment]
From the page you linked:

> Note: Popover only provides the overlay itself. It should be combined with [Dialog](https://react-spectrum.adobe.com/react-aria/Dialog.html) to create fully accessible popovers. Other overlays such as menus may also be placed in a popover.

Our `Menu` component fulfills the same requirements as `Dialog`, so you don't need to use `Dialog` in that case. See examples https://react-spectrum.adobe.com/react-aria/Menu.html#menu

So the answer is, it depends on what you're putting inside the Popover, and it would be an improvement to document what is required of the child of the Popover.

Things I would check to make sure it's working for your case are that focus cannot leave the `Popover` content, focus is moved inside automatically, and focus is restored back to the trigger when closed.
Also, that an accessible message or label pertaining to the `Popover` content as a whole is read when entering the `Popover`. This is a looser requirement because it depends on the content again.
Check that the content has an appropriate role for what you're trying to accomplish.
Ensure that any keyboard interactions, such as Escape to close in some situations, are working.

Or, just use `Dialog` or `Menu` as those take care of all that for you.
